### ISSUE TITLE
 Allow cross-origin resource sharing for EidSuggest

### ISSUE THREAD

**Issue by dkd-friedrich** [TYPO3-Solr/ext-solr#523]
Patrick Broens wrote:

> EidSuggest does not allow cross domain Ajax. This can be done by enabling cross-origin resource sharing (CORS) using the right headers like "Header set Access-Control-Allow-Origin "*"" for Apache, but is not always desired.
> 
> It would be better to use "JSON with padding" (jsonp) in the ajax call by changing the dataType to 'jsonp', add a callback parameter and let the eID script return the proper notation for jsonp. JSON with padding can be used for cross-origin resource sharing by nature, so no extra settings, like the header, are involved.
> 
> This is needed because SuggestFormModifier is using t3lib_div::getIndpEnv('TYPO3_SITE_URL') for the javascript variable tx_solr_suggestUrl. When the site is using multiple domains, this cached value does not have to be the domain you are using to see the website.

---

https://forge.typo3.org/issues/65115

Effort: 8
Value: 2

**Comment by timohund** [comment]
This is possible with the new suggest since version 8 (https://github.com/TYPO3-Solr/ext-solr/pull/1638). Therefore this issue can be closed
